File: src/main/java/co/herod/adbwrapper/rx/ResultChangeFixedDurationTransformer.java
package co.herod.adbwrapper.rx;

import io.reactivex.Observable;
import org.jetbrains.annotations.NotNull;

import java.util.concurrent.TimeUnit;

public class ResultChangeFixedDurationTransformer extends FixedDurationTransformer {

    public ResultChangeFixedDurationTransformer() {
        super(2, TimeUnit.DAYS);
    }

    @Override
    public Observable<String> apply(@NotNull Observable<String> upstream) {
        return super.apply(upstream.repeat()).distinctUntilChanged();
    }
}

File: adb-kotlin/src/main/java/co/herod/adbwrapper/rx/ResultChangeFixedDurationTransformer.kt
package co.herod.adbwrapper.rx

import io.reactivex.Observable

import java.util.concurrent.TimeUnit

class ResultChangeFixedDurationTransformer<T> : FixedDurationTransformer<T>(2, TimeUnit.DAYS) {

    override fun apply(upstream: Observable<T>): Observable<T> {
        return super.apply(upstream.repeat()).distinctUntilChanged()
    }
}
